netdata/netdata
# Additional CO-RE code (eBPF.plugin)

##### Summary
Add eBPF CO-RE codes for threads that only used legacy code to achieve goals:
- Simplify packaging for most recent Linux distributions
- Reduce overload on operate system.

##### Test Plan
1. Compile this branch
2. Enable threads inside `/etc/netdata/ebpf.d.conf`:
```sh
[ebpf programs]
    disk = yes
    filesystem = yes
    hardirq = yes
    mdflush = yes
```
3. Run Netdata and verify that number of threads running `CO-RE` increased http://localhost:19999/api/v1/data?chart=netdata.ebpf_load_methods&after=-10.

##### Additional Information
This PR was tested on:

| Linux Distribution | Kernel version | Query | 
|----------------------------|-----------------------|----------|
|Slackware Current | 6.1.28 | [slackware_6_1.txt](https://github.com/netdata/netdata/files/11521312/slackware_6_1.txt) |
| Arch Linux | 6.3.2-arch1-1 | [alma_6_3.txt](https://github.com/netdata/netdata/files/11521313/alma_6_3.txt) |
| Ubuntu 22.04 | 5.15.0-56-generic | [ubuntu_5_15.txt](https://github.com/netdata/netdata/files/11521429/ubuntu_5_15.txt) | 
| Alma 9 | 5.14.0-284.11.1.el9_2.x86_64 | [alma_5_14.txt](https://github.com/netdata/netdata/files/11522720/alma_5_14.txt) | 
|Debian 11 | 5.10.0-22-amd64 | [debian_5_10.txt](https://github.com/netdata/netdata/files/11523123/debian_5_10.txt) | 
| Oracle 8.6 | kernel-uek-5.4.17-2136.318.7.2 | [oracle_5_4.txt](https://github.com/netdata/netdata/files/11523176/oracle_5_4.txt) | 
|Slackware Current | 5.4.210 |  [slackware_5_4_210.txt](https://github.com/netdata/netdata/files/11523256/slackware_5_4_210.txt) |
|Alma 8.6 | 4.18.0-425.19.2.el8_7.x86_64 | [alma_4_18.txt](https://github.com/netdata/netdata/files/11523336/alma_4_18.txt) | 
|Slackware Current | 4.14.290 | [slackware_4_14_290.txt](https://github.com/netdata/netdata/files/11523255/slackware_4_14_290.txt) | 

You can get all logs using [this](https://github.com/netdata/netdata/files/11526105/all.zip) link.

<details> <summary>For users: How does this change affect me?</summary>
Describe the PR affects users: 
- Which area of Netdata is affected by the change? eBPF.plugin
- Can they see the change or is it an under the hood? If they can see it, where? They can access netdata monitoring and take a look into eBPF charts.
- How is the user impacted by the change? Reduce overload on OS with latest kernels.
- What are there any benefits of the change?  Improve plugin performance.
</details>
